lib/appium.js

```diff
@@ -176,15 +178,26 @@ class AppiumDriver extends BaseDriver {
       }
     }
 
-    let curSessions;
+    let runningDriversData;
     try {
-      curSessions = this.curSessionDataForDriver(InnerDriver);
+      runningDriversData = this.curSessionDataForDriver(InnerDriver);
     } catch (e) {
       throw new errors.SessionNotCreatedError(e.message);
     }
 
+    let innerSessionId, dCaps;
     let d = new InnerDriver(this.args);
-    let [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, curSessions);
+    try {
+      if (_.isArray(this.pendingDrivers[InnerDriver.name])) {
+        this.pendingDrivers[InnerDriver.name].push(d);
+      } else {
+        this.pendingDrivers[InnerDriver.name] = [d];
+      }
+      const otherPendingDriversData = this.pendingDrivers[InnerDriver.name].filter((drv) => d !== drv).map((drv) => drv.driverData);
```

Comment:
would I be correct in assuming that this line of code is attempting to find all the pending sessions other than the one we just added to the hash?

if so, why not just discard the `filter` and do the `map` _before_ we push the current driver? i.e.,

```js
try {
  if (!this.pendingSessions[InnerDriver.name]) {
    this.pendingSessions[InnerDriver.name] = [];
  } 
  const otherPendingSessionsData = this.pendingSessions[InnerDriver.name].map(d => d.driverData);
  this.pendingSessions[InnerDriver.name].push(d);
  ...
}
```

etc